api/src/main/java/jakarta/mail/Header.java

```diff
@@ -67,4 +67,36 @@ public String getName() {
     public String getValue() {
 	return value;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((name == null) ? 0 : name.hashCode());
+        result = prime * result + ((value == null) ? 0 : value.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Header other = (Header) obj;
+        if (name == null) {
```

Comment:
Could `java.util.Objects.equals(name, other.name) && java.util.Objects.equals(value, other.value)` be used instead?

Reply:
For the equals I have to verify first that `obj` is not null and it is the same class before using `Objects.equals`.